pkg/db/db_local/internal.go

```diff
@@ -14,24 +16,97 @@ import (
 	"github.com/turbot/steampipe/sperr"
 )
 
-/**
-
-Query to get functions:
-SELECT
-    p.proname AS function_name
+// dropLegacyInternal looks for a schema named 'internal'
+// which has a function called 'glob' and maybe a table named 'connection_state'
+// and drops it
+func dropLegacyInternal(ctx context.Context, conn *pgx.Conn) error {
+	utils.LogTime("db_local.dropLegacyInternal start")
+	defer utils.LogTime("db_local.dropLegacyInternal end")
+
+	log.Println("[TRACE] querying for legacy 'internal' schema")
+
+	legacySchemaCountQuery := `
+WITH 
+internal_functions AS (
+		SELECT
+			COALESCE(STRING_AGG(DISTINCT(p.proname),','),'') as function_names
+		FROM
+			pg_proc p
+			LEFT JOIN pg_namespace n ON p.pronamespace = n.oid
+		WHERE
+			n.nspname = $1
+),
+internal_tables AS (
+		SELECT 
+				COALESCE(STRING_AGG(DISTINCT(table_name),','),'') as table_names
+		FROM 
+				information_schema.tables 
+		WHERE 
+				table_schema = $1
+)
+SELECT 
+		internal_functions.function_names, 
+		internal_tables.table_names 
 FROM
-    pg_proc p
-    LEFT JOIN pg_namespace n ON p.pronamespace = n.oid
-WHERE
-    n.nspname = 'functionSchema'
-ORDER BY
-    function_name;
+		internal_functions 
+INNER JOIN
+		internal_tables
+		ON true;
+	`
+
+	row := conn.QueryRow(ctx, legacySchemaCountQuery, constants.LegacyInternalSchema)
+
+	var fNames string
+	var tNames string
+	err := row.Scan(&fNames, &tNames)
+	if err != nil {
+		return sperr.WrapWithMessage(err, "could not query legacy 'internal' schema objects: '%s'", constants.LegacyInternalSchema)
+	}
+
+	if len(fNames) == 0 && len(tNames) == 0 {
+		log.Println("[TRACE] could not find any objects in 'internal' - skipping drop")
+		return nil
+	}
+
+	functionNames := strings.Split(fNames, ",")
+	tableNames := strings.Split(tNames, ",")
+
+	log.Println("[TRACE] dropLegacyInternal: available function names", functionNames)
+	log.Println("[TRACE] dropLegacyInternal: available table names", tableNames)
+
+	expectedFunctions := []string{"glob"}
+	expectedTables := []string{"connection_state", constants.ConnectionStateTable}
 
-**/
+	unexpectedFunctions := helpers.StringSliceDiff(functionNames, expectedFunctions)
+	unexpectedTables := helpers.StringSliceDiff(tableNames, expectedTables)
```

Comment:
how does this handle `connection_state` not being there, or both `connection_state` AND `steampipe_connection_state` being there

Reply:
I don;t really like that StringSliceDiff function - it's not clear what it does or returns.

Instead, can't we do what I suggested in standup: 
```
	expectedFunctions := map[string]bool{
		"glob": true,
	}
	expectedTables := map[string]bool{
		"connection_state":             true, // legacy table name
		constants.ConnectionStateTable: true,
	}

	for _, f := range functions {
		if !expectedFunctions[f]{
			return nuil
		}
	} 
	for _, t := range tables {
		if !expectedTables[t]{
			return nuil
		}
	}
```

Reply:
or something